crates/astria-core/src/primitive/v1/asset.rs

```diff
@@ -83,6 +88,48 @@ impl Denom {
     pub fn to_base_denom(&self) -> Self {
         Self::from_base_denom(&self.base_denom)
     }
+
+    /// Create a new [`Denom`] with the given prefix removed.
+    ///
+    /// It also ensures the resulting denom does not have a slash at the beginning.
+    /// For example, if the denom is `prefix/base`, and the prefix to remove is `prefix`,
+    /// then the resulting denom will be `base`.
+    ///
+    /// # Errors
+    ///
+    /// - if the denom does not have the given prefix.
+    pub fn remove_prefix(&self, prefix: &str) -> Result<Self, InvalidPrefixToRemove> {
+        let prefix_to_remove = prefix.trim_end_matches('/');
+        if prefix_to_remove == self.prefix {
+            return Ok(self.to_base_denom());
+        }
+
+        let new_prefix = self
+            .prefix
+            .clone()
+            .strip_prefix(prefix_to_remove)
+            .ok_or_else(|| InvalidPrefixToRemove {
+                prefix: prefix.to_string(),
+                denom: self.to_string(),
+            })?
+            .trim_start_matches('/')
```

Comment:
Should we maybe check at this point that the stripped prefix is either empty or starts with `/` and return an error if not?  Just now this would allow stripping e.g. `p0/p` from `p0/p1` leaving `1` which probably isn't desirable?

Either way - I guess a test for that case would be useful.